src/shared/components/ACMIDMNudgeModal/ACMIDMNudgeModal.tsx

```diff
@@ -94,9 +68,29 @@ const ACMIDMNudgeModal: FC<UserProps & UiStateProps & RouteComponentProps> = ({
 		const isOnAccountLinkingPage = location.pathname.includes(APP_PATH.SETTINGS_LINKS.route);
 
 		if (user && !isOnAccountLinkingPage && (!isPupil || (isPupil && !isOnAssignmentPage))) {
-			fetchProfilePreference();
+			const profilePreferences =
+				(await ProfilePreferencesService.fetchProfilePreference(
+					getProfileId(user),
+					ProfilePreferenceKey.DoNotShow
+				)) || [];
+
+			const hasVlaamseOverheidLinked =
+				!!user &&
+				(hasIdpLinked(user, 'VLAAMSEOVERHEID__SUB_ID') ||
+					hasIdpLinked(user, 'VLAAMSEOVERHEID__ACCOUNT_ID'));
+			const profileIsComplete = user && isProfileComplete(user);
+
+			setShowNudgingModal(
+				!profilePreferences.length && !hasVlaamseOverheidLinked && profileIsComplete
+			);
+		} else {
+			setShowNudgingModal(false);
 		}
-	}, [fetchProfilePreference, user, isPupil, location, setShowNudgingModal]);
+	}, [user, isPupil, location, setShowNudgingModal]);
+
+	useEffect(() => {
+		updateShowNudgingModal();
+	}, [updateShowNudgingModal]);
```

Comment:
This is functionally identical to using one effect because the dependency array`[updateShowNudgingModal]` equals `[user, isPupil, location, setShowNudgingModal]`.

Reply:
yes, but then you can't use await inside the useEffect, because it doesn't accept a handler that returns a promise